packages/flutter_plugin_android_lifecycle/android/proguard.txt

```diff
@@ -0,0 +1,4 @@
+# The point of this package is to specify that a dependent plugin intends to
+# use the AndroidX lifecycle classes. Make sure no R8 heuristics shrink classes
+# brought in by the embedding's pom.
+-keep class androidx.lifecycle.DefaultLifecycleObserver
```

Comment:
Interesting. This is definitely a better fix than https://github.com/flutter/plugins/pull/3766. I'm surprised that by slightly changing the code, this rule isn't needed